src/nativeMain/kotlin/api/EclairClient.kt

```diff
@@ -22,6 +24,9 @@ class EclairClientBuilder : IEclairClientBuilder {
 
 interface IEclairClient {
     suspend fun getInfo(): Either<ApiError, String>
+    suspend fun connectUri(uri: String): Either<ApiError, String>
+    suspend fun connectNodeId(nodeId: String): Either<ApiError, String>
```

Comment:
You shouldn't use different methods, it would be much cleaner to use:

```suggestion
    sealed class ConnectionTarget {
        data class Uri(val uri: String) : ConnectionTarget()
        data class NodeId(val nodeId: String) : ConnectionTarget()
    }

    suspend fun connect(target: ConnectionTarget): Either<ApiError, String>
```

Reply:
Made these changes [here](https://github.com/ACINQ/eclair-cli/pull/9/commits/d97c1a99432683b923f6951751bbd038bee13139)